src/NexusMods.App.UI/Pages/Sorting/LoadOrder/TreeDataGrid/LoadOrderComponents.cs

```diff
@@ -0,0 +1,92 @@
+using System.ComponentModel;
+using JetBrains.Annotations;
+using NexusMods.Abstractions.UI;
+using NexusMods.App.UI.Controls;
+using R3;
+using static NexusMods.App.UI.Pages.Sorting.LoadOrderComponents;
+
+namespace NexusMods.App.UI.Pages.Sorting;
+
+public static class LoadOrderComponents
+{
+    public sealed class IndexComponent : ReactiveR3Object, IItemModelComponent<IndexComponent>, IComparable<IndexComponent>
+    {
+        private readonly ValueComponent<int> _index;
+        private readonly ValueComponent<string> _displaySortIndex;
+
+        public ReactiveCommand<Unit> MoveUp { get; } = new();
+        public ReactiveCommand<Unit> MoveDown { get; } = new();
+
+        public IReadOnlyBindableReactiveProperty<int> SortIndex => _index.Value;
+        public IReadOnlyBindableReactiveProperty<string> DisplaySortIndex => _displaySortIndex.Value;
+
+        public IndexComponent(ValueComponent<int> index, ValueComponent<string> displaySortIndex)
+        {
+            _index = index;
+            _displaySortIndex = displaySortIndex;
+        }
+
+        public int CompareTo(IndexComponent? other)
+        {
+            // Data is sorted by the Adapter, not the treeDataGrid, this should not be called
+            throw new NotSupportedException();
+        }
+
+        internal static string IndexToOrdinal(int sortIndex)
+        {
+            var displayIndex = sortIndex + 1;
+            var suffix = displayIndex switch
+            {
+                11 or 12 or 13 => "th",
+                _ when displayIndex % 10 == 1 => "st",
+                _ when displayIndex % 10 == 2 => "nd",
+                _ when displayIndex % 10 == 3 => "rd",
+                _ => "th"
+            };
+            return $"{displayIndex}{suffix}";
+        }
```

Comment:
What's this supposed to be?

Reply:
You probably want this: https://github.com/Humanizr/Humanizer?tab=readme-ov-file#ordinalize

Reply:
Yeah, that was the previous code that Simon setup for the DisplayIndex. Good to see a Humanizer option for this